openidl-common-lib/middlewares/DBService/DBManagerFactory.js

```diff
@@ -1,39 +1,35 @@
 const log4js = require('log4js');
 const CloudantDBManager = require('./cloudantDBManager');
 const MongoDBManager = require('./mongoDBManager');
+const PostgresDBManager = require('./postgresDBManager');
+
 const logger = log4js.getLogger('DBManagerFactory');
 logger.level = process.env.LOG_LEVEL || 'debug';
 
 class DBManagerFactory {
-    constructor() {
+  async getInstance(options, dbType) {
+    logger.info(`Inside DBManagerFactory getInstance(${dbType}): ${JSON.stringify(options)}`);
+    let databaseType = dbType;
+    if (databaseType === undefined) {
+      databaseType = options.defaultDbType;
     }
-    async getInstance(options) {
-        if (!options || !options.persistentStore) {
-            logger.error('Database config not found!!');
-        }
-        logger.info('Inside DBManagerFactory getInstance');
-        logger.info('Persistent store selected', options.persistentStore);
-        try {
-            switch (options.persistentStore) {
-                case "mongo":
-                    let mongoDBInstance = new MongoDBManager(options);
-                    let mongoDBName = options.mongodb;
-                    let simpleURI = options.simpleURI
-                    if (simpleURI) {  // use the simple db connection
-                        mongoDBInstance.initSimpleMongoDBConnection(mongoDBName, simpleURI)
-                    } else {  // use the cloud db connection
-                        await mongoDBInstance.initMongoDBConnection(mongoDBName);
-                    }
-                    return mongoDBInstance;
-                case "cloudant":
-                    return new CloudantDBManager(options)
-                default:
-                    logger.error("Incorrect Usage of database type. Refer README for more details");
-                    break;
-            }
-        } catch (error) {
-            throw error;
-        }
+
+    try {
+      switch (databaseType) {
+        case 'mongo':
+          const mongoDBInstance = new MongoDBManager(options.mongo);
+          return mongoDBInstance;
+        case 'cloudant':
+          return new CloudantDBManager(options.cloudant);
+        case 'postgres':
+          return new PostgresDBManager(options.postgres);
+        default:
+          logger.error(`Incorrect Usage of database type: ${dbType}. Refer README for more details`);
+          throw Error(`Unknown DB type: ${dbType}`);
+      }
+    } catch (error) {
```

Comment:
There is no point of having the Try/Catch block if we are just going to rethrow the error.